### PR TITLE
Improved macro settings

### PR DESCRIPTION
**Pull request by breiler**
Attempt to fix a bunch of issues with the macro panel:
#980 - deletion is now made once the user selects save
#454 - updates are now synced with the action panel upon save
#889 - added functions to reorder macros

In the classic edition the settings panel can be opened from the menu:
image

In the platform edition an menu action has been added to deep link to the macros settings. Macros are only saved once Apply or OK is selected.
image

@DanielWeigl could you make sure I got your changes merged right?

### CODE REVIEW

**Comment by DanielWeigl** [comment]
Hi, looks generally good to me.

Just noticed the order in the menu is off (havent looked at the code):
grafik

grafik

**Comment by DanielWeigl** [comment]
Ah, also just noticed one missed merge.  Ive fixed it on top oif your changes, just cherry-pick it or merge it: https://github.com/DanielWeigl/Universal-G-Code-Sender/commit/e5fe6bcf907a9c9f41568796d462718187f30a37

**Comment by breiler** [comment]
Thanks @DanielWeigl for taking the time testing the PR! The order of the macros should now be correct.

**Comment by breiler** [comment]
@winder have you had the time to look through these changes? Does the UI stuff look ok to you?

File: ugs-platform/ugs-platform-ugscore/src/main/java/com/willwinder/ugs/nbp/core/actions/EditMacrosAction.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        putValue("iconBase", ICON_BASE);
+        putValue(SMALL_ICON, ImageUtilities.loadImageIcon(ICON_BASE, false));
+        putValue("menuText", LocalizingService.EditMacrosTitle);
+        putValue(NAME, LocalizingService.EditMacrosTitle);
+        setEnabled(true);
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        OptionsDisplayer.getDefault().open(MacrosOptionsPanelController.KEYWORDS_CATEGORY);
```
Comment (winder) [inline ugs-platform/ugs-platform-ugscore/src/main/java/com/willwinder/ugs/nbp/core/actions/EditMacrosAction.java]: Nice touch :+1:

**Comment by BallscrewBob** [comment]
Small issue.
Seems related to this change but not 100% sure.
If I export my settings and macros from ..
Product Version: Universal Gcode Platform 201609300101
Java: 1.8.0_181; Java HotSpot(TM) 64-Bit Server VM 25.181-b13
Runtime: Java(TM) SE Runtime Environment 1.8.0_181-b13
System: Windows 7 version 6.1 running on amd64; Cp1252; en_US (ugsplatform)

Then import them into the latest nightly the whole program stops.

**Comment by breiler** [comment]
@BallscrewBob I couldn't reproduce the error. Could you post the log message from:
* Windows 7: C:\Users\username\AppData\Roaming\ugsplatform\2.0-SNAPSHOT\var\log\messages.log
* Mac: ~/Library/Application Support/ugsplatform\2.0-SNAPSHOT/var/log/messages.log
* Linux: ~/.ugsplatform/2.0-SNAPSHOT/var/log/messages.log

**Comment by vMeph** [comment]
just one question where is now where we can download the latest  changes tin  UGS plaform, before there was a site for download all the time changes made in a zip file  cant find it anymore

**Comment by BallscrewBob** [comment]
I have a split OOB install and the file you reference does not seem to exist anywhere on the computer.
Does it get cleared after a total shutdown of UGS ?

**Comment by BallscrewBob** [comment]
Attaching a file from CRASHDUMP

**Comment by breiler** [comment]
No the directory shouldn't get cleared. Try opening the about-dialog and get the config-directory shown in the screenshot here: http://winder.github.io/ugs_website/guide/troubleshooting/#property-files

Go up one directory from that and there should be a log directory.

Tried opening your crashdump with no success and I am not able to reproduce the error... =/

Tried starting a virtual machine with Windows 10 and installed the oldest version of UGS that I could find, then upgrade to the latest.

Tried importing your settings into the latest version of UGS.

If this causes any more problems, please create a new issue and will dig some more.

**Pull request merged**